### PR TITLE
(George_Roumieh-w4-javascript)Finished week4 homework

### PR DESCRIPTION
**Pull request by george2621**

### CODE REVIEW

File: 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js [hunk 1]
```diff
@@ -10,16 +10,9 @@ Let's rewrite it (or _refactor_ it, as experienced developers would call it):
...
-  const newNumbers = [];
-  for (let i = 0; i < numbers.length; i++) {
-    if (numbers[i] % 2 === 0) {
-      newNumbers.push(numbers[i] * 2);
-    }
-  }
-  return newNumbers;
-}
+const doubleEvenNumbers = (numbers) => {
+  return numbers.filter((num) => num % 2 === 0).map((num) => num * 2);
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex1-doubleEvenNumbers.test.js:14]: 👍

File: 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js [hunk 2]
```diff
@@ -31,8 +31,14 @@ const mondayTasks = [
 
 const hourlyRate = 25;
 
-function computeEarnings(/* TODO parameter(s) go here */) {
+function computeEarnings(tasksArray, rate) {
   // TODO complete this function
+  const taskDuration = tasksArray.map((task) => task.duration / 60);
+  const total = taskDuration.reduce(
+    (total, cur) => (total = total + cur * rate),
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex2-mondaysWorth.test.js:38]: 👍

File: 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js [hunk 3]
```diff
@@ -26,26 +26,31 @@ const fruitBasket = [
 ];
 
 // ! Function under test
-function sanitizeFruitBasket(/* TODO parameter(s) go here */) {
+function sanitizeFruitBasket(fruitsArray, allergicFruit) {
   // TODO complete this function
+  const filteredFruits = fruitsArray.filter((fruit) => fruit !== allergicFruit);
+  return filteredFruits;
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex3-lemonAllergy.test.js:32]: 👍

File: 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js [hunk 4]
```diff
@@ -18,12 +18,18 @@ function createObservable() {
...
     subscribe: function (subscriber) {
       // TODO complete this function
+      subscribers.push(subscriber);
     },
     notify: function (message) {
       // TODO complete this function
+      subscribers.forEach(
+        (subscriber) =>
+          subscriber(message) + console.log(`(${subscriber.name} subscriber) `)
+      );
```
Comment (yash-kapila) [inline 1-JavaScript/Week4/homework/ex4-observable/ex4-observable.js:28]: 👍

**Review by yash-kapila — COMMENTED** [review]
Hi @george2621 - I've finished reviewing the homework. Can you please have a look at the comments?

**Comment by george2621** [comment]
@yash-kapila Thanks for the comment and review.

**Pull request closed without merging**